### PR TITLE
PWN-5330 - Fix last restore edge cases

### PR DESCRIPTION
**Pull request by davran1001**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-5330

## Description of Work

Finish all restore edge cases

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/auth/interactor/restore/UserRestoreInteractor.kt [hunk 1]
```diff
@@ -177,23 +177,49 @@ class UserRestoreInteractor(
...
+
+        // if restore is FAILED we throw error thast SHARES ARE NOT MATCH
+        // otherwise Restore is SUCCESSFULL
+        return when (result) {
+            is RestoreUserResult.RestoreFailure -> {
+                RestoreUserResult.RestoreFailure.DevicePlusSocialOrSocialPlusCustom(
+                    RestoreUserException("Shares are not match")
+                )
+            }
+            else -> result
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/interactor/restore/UserRestoreInteractor.kt:197]: {}

File: app/src/main/java/org/p2p/wallet/auth/interactor/restore/UserRestoreInteractor.kt [hunk 2]
```diff
@@ -177,23 +177,49 @@ class UserRestoreInteractor(
...
             result = tryRestoreUser(OnboardingFlow.RestoreWallet.SocialPlusCustomShare)
         }
-        return result
+
+        // if restore is FAILED we throw error thast SHARES ARE NOT MATCH
+        // otherwise Restore is SUCCESSFULL
+        return when (result) {
+            is RestoreUserResult.RestoreFailure -> {
+                RestoreUserResult.RestoreFailure.DevicePlusSocialOrSocialPlusCustom(
+                    RestoreUserException("Shares are not match")
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/interactor/restore/UserRestoreInteractor.kt:194]: Shares device+social or social+custom are not match

Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/auth/interactor/restore/UserRestoreInteractor.kt:194]: same with all pieces of code

File: app/src/main/java/org/p2p/wallet/auth/repository/RestoreUserExceptionHandler.kt [hunk 3]
```diff
@@ -79,7 +103,7 @@ class RestoreUserExceptionHandler(
                     )
                 )
             }
-            else -> RestoreFailureState.ToastError("Error on restore Social + Custom Share")
+            else -> RestoreFailureState.LogError(result.message ?: result.exception.message.orEmpty())
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/repository/RestoreUserExceptionHandler.kt:106]: LogError Never used

**Review by kabramovp2p — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/auth/interactor/restore/UserRestoreInteractor.kt (hunk 2 above)
Comment (davran1001) [reply app/src/main/java/org/p2p/wallet/auth/interactor/restore/UserRestoreInteractor.kt:194]: Yes handling the same, i think it's not so bad, we can change it after. Now our target to pass this feature , ASAP))

**Review by gslevinkov — APPROVED** [review]

**Pull request merged**